packages/lab/src/dialog/Dialog.tsx

```diff
@@ -89,49 +107,47 @@ export const Dialog = forwardRef<HTMLDivElement, DialogProps>(function Dialog(
     [dialogId, status]
   );
 
+  if (!open) {
```

Comment:
This seems weird, I think FloatingComponent is normally our "source" of whether something is open or not. This would mean FloatingComponent is only every mounted when open. Not sure of the side effects on desktop support.

Reply:
switched to a conditional render